dashboard/lib/queries/courses.rb

```diff
@@ -15,7 +15,7 @@ class Queries::Courses
   def self.get_course_context(unit_name)
     unit = Unit.get_from_cache(unit_name)
     return nil unless unit
-    unit_group_unit = UnitGroupUnit.where(script_id: unit.id).first
+    unit_group_unit = UnitGroupUnit.get_with_unit_from_cache(unit.id).first
```

Comment:
unit.unit_group_units should already be in the cached response from Unit.get_from_cache: https://github.com/code-dot-org/code-dot-org/blob/c6b3e53b868041e274f75a2e2a86d4882652a567/dashboard/app/models/unit.rb#L98

based on that, my understanding is that this should accomplish the goal of eliminating the database request:
```
    unit_group_unit = unit.unit_group_units.first
```

my recommendation would be to write a test which uses `assert_queries` for the codepath you care about, and then use it to confirm whether each approach succeeds in reducing the number of queries.

Reply:
I wrote did some testing and you are right! `unit.unit_group_units` does get cached. I have removed that helper function.